### PR TITLE
Implement get all user donations

### PR DESCRIPTION
**Pull request by dplamenov**
Implement new path for get all user donations.

New path: 
new path

### CODE REVIEW

File: apps/api/src/account/account.controller.ts [hunk 1]
```diff
@@ -42,4 +43,9 @@ export class AccountController {
   adminRole() {
     return { status: 'OK' }
   }
+
+  @Get('donations')
+  async userDonations(@AuthenticatedUser() user: KeycloakTokenParsed) {
+    return await this.accountService.getDonationsByUser(user.sub as string)
+  }
```
Comment (kachar) [inline apps/api/src/account/account.controller.ts:50]: Maybe this part is more appropriate to go to `donations.controller.ts` as it's not really part of user profile

**Comment by dplamenov** [comment]
I moved `userDonations` method in `DonationsController`. Now path is: `api/donation/user-donations`
Should I move `getDonationsByUser` method to `DonationsService`?

File: apps/api/src/donations/donations.controller.spec.ts [hunk 2]
```diff
@@ -26,6 +27,7 @@ describe('DonationsController', () => {
           provide: STRIPE_CLIENT_TOKEN,
           useValue: jest.fn(),
         },
+        AccountService
```
Comment (kachar) [inline apps/api/src/donations/donations.controller.spec.ts:30]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/account/account.service.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import { Injectable } from '@nestjs/common'
+import { PrismaService } from '../prisma/prisma.service'
 
 @Injectable()
-export class AccountService {}
+export class AccountService {
+  constructor() { }
```
Comment (kachar) [inline apps/api/src/account/account.service.ts:6]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/account/account.service.spec.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+import { PrismaService } from '../prisma/prisma.service'
 import { AccountService } from './account.service'
 
 describe('AccountService', () => {
   let service: AccountService
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [AccountService],
+      providers: [AccountService, PrismaService],
```
Comment (kachar) [inline apps/api/src/account/account.service.spec.ts:10]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/account/account.module.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { Module } from '@nestjs/common'
+import { PrismaService } from '../prisma/prisma.service'
 import { AccountController } from './account.controller'
 import { AccountService } from './account.service'
 
 @Module({
   controllers: [AccountController],
-  providers: [AccountService],
+  providers: [AccountService, PrismaService],
```
Comment (kachar) [inline apps/api/src/account/account.module.ts:8]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/account/account.controller.ts [hunk 6]
```diff
@@ -4,11 +4,9 @@ import { AuthenticatedUser, Public, RoleMatchingMode, Roles } from 'nest-keycloa
 
 import { KeycloakTokenParsed } from '../auth/keycloak'
 
-import { AccountService } from './account.service'
-
 @Controller('account')
 export class AccountController {
-  constructor(private readonly accountService: AccountService) {}
+  constructor() {}
```
Comment (kachar) [inline apps/api/src/account/account.controller.ts:9]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/account/account.controller.spec.ts [hunk 7]
```diff
@@ -8,7 +9,7 @@ describe('AccountController', () => {
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
       controllers: [AccountController],
-      providers: [AccountService],
+      providers: [AccountService, PrismaService],
```
Comment (kachar) [inline apps/api/src/account/account.controller.spec.ts:12]: You can undo changes in this file as they are no longer relevant

File: apps/api/src/donations/donations.controller.ts [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
 import { Body, Controller, Get, Post } from '@nestjs/common'
 
 import { DonationsService } from './donations.service'
 import { CreateSessionDto } from './dto/create-session.dto'
+import { KeycloakTokenParsed } from '../auth/keycloak'
+import { AccountService } from '../account/account.service'
 
 @Controller('donation')
 export class DonationsController {
-  constructor(private readonly paymentsService: DonationsService) {}
```
[the remark this answers is not in the document]
Comment (kachar) [reply apps/api/src/donations/donations.controller.ts:9]: Totally, feel free to rename it

Maybe in the past the service had different name and the local variable is not changed after it's been renamed

**Review by kachar — APPROVED** [review]
@dplamenov Awesome contribution!

The final version is so concise and specific to the use-case that we want to improve! :rocket: :moon: 

Thanks for the energy spent in iterations to achieve that :zap:

**Pull request merged**